bpmn-model/src/main/java/io/zeebe/model/bpmn/validation/zeebe/EventBasedGatewayValidator.java

```diff
@@ -103,4 +109,11 @@ private boolean isValidEvent(final IntermediateCatchEvent event) {
         .filter(e -> e instanceof MessageEventDefinition)
         .map(MessageEventDefinition.class::cast);
   }
+
+  private boolean succeedingNodesOnlyHaveEventBasedGatewayAsIncomingFlows(
+      EventBasedGateway element) {
+    return element.getSucceedingNodes().stream()
+        .flatMap(flowNode -> flowNode.getPreviousNodes().stream())
+        .allMatch(flowNode -> flowNode.equals(element));
```

Comment:
I think `.allMatch(flowNode -> flowNode.equals(element))` could be written `.allMatch(element::equals)` but I'll let you decide, it's pretty much micro optimization at that point so meh